DOCSUP-1394 (#139)
BayoNet/ClickHouse: docs/en/operations/system-tables.md

If the storage policy contains more then one volume, then information for each volume is stored in the individual row of the table.

##system.roles {#system_tables-roles}

> Contains information about [role](../operations/access-rights.md#role-management) that can be granted to a user account.

Reviewer: ```suggestion
Contains information about configured [roles](../operations/access-rights.md#role-management).
```

Columns:
- name ([String](../sql-reference/data-types/string.md)) — Role name.
- id ([UUID](../sql-reference/data-types/uuid.md)) — Role ID.

> - storage ([String](../sql-reference/data-types/string.md)) — Storage of roles. The folder path on disk is set in the config file — `access_control_path` parameter.

Reviewer: ```suggestion
- storage ([String](../sql-reference/data-types/string.md)) — Path to the storage of roles. Configured in the `access_control_path` parameter.
```

##system.role_grants {#system_tables-role_grants}

> Contains the role grants for users and roles. To add entries to this table, use 'GRANT role TO user'.

Reviewer: ```suggestion
Contains the role grants for users and roles. To add entries to this table, use `GRANT role TO user`.
```

> Contains the role grants for users and roles. For adds entries to this table use 'GRANT role TO user'.

Reviewer: ```suggestion
Contains the role grants for users and roles. To add entries to this table, use `GRANT role TO user`.
```

Columns:
- user_name ([Nullable](../sql-reference/data-types/nullable.md)([String](../sql-reference/data-types/string.md))) — User name.
- role_name ([Nullable](../sql-reference/data-types/nullable.md)([String](../sql-reference/data-types/string.md))) — Role name.

> - granted_role_name ([String](../sql-reference/data-types/string.md)) — Role name when it is granted to another role. In order for a role granted a different role, use 'GRANT role1 TO role2'.

Reviewer: ```suggestion
- granted_role_name ([String](../sql-reference/data-types/string.md)) — Name of role granted to the `role_name` role. To grant one role to another one use `GRANT role1 TO role2`.
```

Reviewer: Does the `granted_role_name` is about a role granted to another role only? I suppose that we talk about the role granted to a user account or a role.

> - granted_role_is_default ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Granted role by default. Possible values:

Reviewer: ```suggestion
- granted_role_is_default ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Flag that shows whether `granted_role` is a default role. Possible values:
```

>     -   1 — The role has default privileges.

Reviewer: ```suggestion
    -   1 — `granted_role` is a default role.
```

>     -   0 — The role without default privileges. 

Reviewer: ```suggestion
    -   0 — `granted_role` is not a default role.
```

- with_admin_option ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role with administrator options. Possible values:
    -   1 — The role has administrator options.
    -   0 — The role without administrator options. 

##system.current_roles {#system_tables-current_roles}

> Contains the current roles of the current user, it is a different table for different users (like system.settings), 'SET ROLE' changes the contents of this table.

Reviewer: ```suggestion
Contains active roles of a current user. `SET ROLE` changes the contents of this table.
```

Columns:
 - role_name ([String](../sql-reference/data-types/string.md))) — Role name.

>  - with_admin_option ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role with administrator options.

Reviewer: Isn't it a flag that shows whether the role has `ADMIN OPTION`?

>  - is_default ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role by default.

Reviewer: This is the flag too.

##system.enabled_roles {#system_tables-enabled_roles}

> Contains: a) current roles of the current user + b) granted roles for a) + granted-roles for b)+ ..., i.e. in fact these are all active roles at the moment.

Reviewer: Didn't understand at all.

Author: Fixed

Columns:
- role_name ([String](../sql-reference/data-types/string.md))) — Role name.

> - with_admin_option ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role with administrator options.

Reviewer: Flag?

> - is_current ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role of current user.

Reviewer: Flag?

> - is_default ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Role by default.

Reviewer: Flag?

Author: Fixed

> - `is_current` ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Flag that shows whether `enabled_role` is a role of current user.

Reviewer: ```suggestion
- `is_current` ([UInt8](../sql-reference/data-types/int-uint.md#uint-ranges)) — Flag that shows whether `enabled_role` is a current role of a current user.
```
